dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/InstrumentedSslConnectionSocketFactory.java

```diff
@@ -101,106 +72,21 @@ public Socket createSocket(HttpContext _context) {
-            }
-            // Run this under a doPrivileged to support lib users that run under a SecurityManager this allows granting
-            // connect permissions only to this library
-            try {
-                AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
-                    doConnect(
-                            sock,
-                            remoteAddress,
-                            Timeout.defaultsToDisabled(connectTimeout).toMillisecondsIntBound());
-                    return null;
-                });
-            } catch (PrivilegedActionException e) {
-                Throwable cause = e.getCause();
-                if (cause instanceof IOException) {
-                    throw (IOException) cause;
-                }
-                throw new SafeIoException("Failed to connect", e, SafeArg.of("clientName", clientName));
-            }
-        } catch (IOException ex) {
-            Closer.closeQuietly(sock);
-            throw ex;
-        }
-        // Setup SSL layering if necessary
-        if (sock instanceof SSLSocket) {
-            SSLSocket sslsock = (SSLSocket) sock;
-            executeHandshake(sslsock, host.getHostName(), attachment);
-            return sock;
-        }
-        return createLayeredSocket(sock, host.getHostName(), remoteAddress.getPort(), attachment, context);
-    }
-
-    private void doConnect(Socket sock, InetSocketAddress remoteAddress, int connectTimeoutMillis) throws IOException {
         boolean success = false;
         long startNanos = System.nanoTime();
         try {
-            sock.connect(remoteAddress, connectTimeoutMillis);
+            super.connectSocket(sock, remoteAddress, connectTimeout, context);
             success = true;
         } finally {
             long durationNanos = System.nanoTime() - startNanos;
             Timer timer = success ? connectTimerSuccess : connectTimerFailure;
             timer.update(durationNanos, TimeUnit.NANOSECONDS);
         }
     }
-
-    private void executeHandshake(SSLSocket sslSocket, String target, Object attachment) throws IOException {
-        TlsConfig tlsConfig = attachment instanceof TlsConfig ? (TlsConfig) attachment : TlsConfig.DEFAULT;
-        if (supportedProtocols != null) {
-            sslSocket.setEnabledProtocols(supportedProtocols);
-        } else {
-            sslSocket.setEnabledProtocols(TLS.excludeWeak(sslSocket.getEnabledProtocols()));
-        }
-        if (supportedCipherSuites != null) {
-            sslSocket.setEnabledCipherSuites(supportedCipherSuites);
-        } else {
-            sslSocket.setEnabledCipherSuites(TlsCiphers.excludeWeak(sslSocket.getEnabledCipherSuites()));
-        }
-        Timeout handshakeTimeout = tlsConfig.getHandshakeTimeout();
-        if (handshakeTimeout != null) {
-            sslSocket.setSoTimeout(handshakeTimeout.toMillisecondsIntBound());
-        }
-
-        prepareSocket(sslSocket);
```

Comment:
Example of how this can be dangerous -- this `prepareSocket` method was deprecated in the weeks since we added this instrumented implementation in favor of an overload with an additional `HttpContext context` parameter. While not breaking, that sort of change can cause us to miss out on bugfixes and new feature support when we replace logic as opposed to wrapping it as this PR does.